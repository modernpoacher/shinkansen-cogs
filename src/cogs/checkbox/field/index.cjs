require('@babel/register')({
  ignore: [
    /node_modules\/(?!shinkansen|@modernpoacher)/
  ]
})

const debug = require('debug')

const log = debug('shinkansen-cogs/cogs/checkbox/field')

log('`shinkansen` is awake')

const {
  default: component // @ts-ignore
} = require('./index.jsx')

module.exports = component
