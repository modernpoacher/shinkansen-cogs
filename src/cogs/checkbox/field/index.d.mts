declare module '#cogs/cogs/checkbox/field' {
  import Field from '#cogs/components/field'

  export default class CheckboxField extends Field {}
}

declare module 'shinkansen-cogs/cogs/checkbox/field' {
  export { default } from '#cogs/cogs/checkbox/field'
}
