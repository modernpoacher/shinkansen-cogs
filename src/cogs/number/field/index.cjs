require('@babel/register')({
  ignore: [
    /node_modules\/(?!shinkansen|@modernpoacher)/
  ]
})

const debug = require('debug')

const log = debug('shinkansen-cogs/cogs/number/field')

log('`shinkansen` is awake')

const {
  default: component
} = require('./index.jsx')

module.exports = component
