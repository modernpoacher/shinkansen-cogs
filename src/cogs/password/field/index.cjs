require('@babel/register')({
  ignore: [
    /node_modules\/(?!shinkansen|@modernpoacher)/
  ]
})

const debug = require('debug')

const log = debug('shinkansen-cogs/cogs/password/field')

log('`shinkansen` is awake')

const {
  default: Field // @ts-expect-error
} = require('./index.jsx')

/**
 *  Exports only default
 */
module.exports = Field
