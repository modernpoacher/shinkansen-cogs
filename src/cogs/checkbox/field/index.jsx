/**
 * CheckboxField component
 */
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { CheckField } from '#cogs/components/field'

export default class CheckboxField extends CheckField {
  getClassName () {
    return classnames(super.getClassName(), 'checkbox')
  }

  render () {
    const {
      checked,
      id,
      name,
      value,
      required = false,
      disabled = false,
      readOnly = false,
      tabIndex,
      accessKey,
      fieldRef
    } = this.props

    const className = this.getClassName()

    if (typeof checked === 'boolean') {
      return (
        <input
          checked={checked}
          id={id}
          name={name}
          value={value}
          required={required}
          disabled={disabled}
          readOnly={readOnly}
          tabIndex={tabIndex}
          accessKey={accessKey}
          onChange={this.handleChange}
          onClick={this.handleClick}
          className={className}
          type='checkbox'
          ref={fieldRef}
        />
      )
    }

    const {
      defaultChecked
    } = this.props

    return (
      <input
        defaultChecked={defaultChecked}
        id={id}
        name={name}
        value={value}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        onChange={this.handleChange}
        onClick={this.handleClick}
        className={className}
        type='checkbox'
        ref={fieldRef}
      />
    )
  }
}

CheckboxField.propTypes = {
  ...CheckField.propTypes,
  value: PropTypes.string.isRequired
}
