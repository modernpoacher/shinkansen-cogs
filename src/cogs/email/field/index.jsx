/**
 * EmailField component
 */
import React from 'react'
import classnames from 'classnames'

import { ValueField } from '#cogs/components/field'

export default class EmailField extends ValueField {
  getClassName () {
    return classnames(super.getClassName(), 'email')
  }

  render () {
    const {
      id,
      name,
      value,
      defaultValue,
      required = false,
      disabled = false,
      readOnly = false,
      tabIndex,
      accessKey,
      placeholder,
      fieldRef
    } = this.props

    const className = this.getClassName()

    return (
      <input
        id={id}
        name={name}
        value={value}
        defaultValue={defaultValue}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onChange={this.handleChange}
        className={className}
        type='email'
        ref={fieldRef}
      />
    )
  }
}

EmailField.propTypes = {
  ...ValueField.propTypes
}
