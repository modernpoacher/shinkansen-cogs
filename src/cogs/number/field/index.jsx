/**
 * NumberField component
 */
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { ValueField } from '#cogs/components/field'

export default class NumberField extends ValueField {
  getClassName () {
    return classnames(super.getClassName(), 'number')
  }

  render () {
    const {
      id,
      name,
      value,
      defaultValue,
      required = false,
      disabled = false,
      readOnly = false,
      tabIndex,
      accessKey,
      placeholder,
      fieldRef
    } = this.props

    const className = this.getClassName()

    return (
      <input
        id={id}
        name={name}
        value={value}
        defaultValue={defaultValue}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onChange={this.handleChange}
        className={className}
        type='number'
        ref={fieldRef}
      />
    )
  }
}

NumberField.propTypes = {
  ...ValueField.propTypes,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  defaultValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}
