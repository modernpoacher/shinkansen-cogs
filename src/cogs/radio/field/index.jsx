/**
 * RadioField component
 */
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { CheckField } from '#cogs/components/field'

export default class RadioField extends CheckField {
  getClassName () {
    return classnames(super.getClassName(), 'radio')
  }

  render () {
    const {
      checked,
      id,
      name,
      value,
      required = false,
      disabled = false,
      readOnly = false,
      tabIndex,
      accessKey,
      fieldRef
    } = this.props

    const className = this.getClassName()

    if (typeof checked === 'boolean') {
      return (
        <input
          checked={checked}
          id={id}
          name={name}
          value={value}
          required={required}
          disabled={disabled}
          readOnly={readOnly}
          tabIndex={tabIndex}
          accessKey={accessKey}
          onChange={this.handleChange}
          onClick={this.handleClick}
          className={className}
          type='radio'
          ref={fieldRef}
        />
      )
    }

    const {
      defaultChecked
    } = this.props

    return (
      <input
        defaultChecked={defaultChecked}
        id={id}
        name={name}
        value={value}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        onChange={this.handleChange}
        onClick={this.handleClick}
        className={className}
        type='radio'
        ref={fieldRef}
      />
    )
  }
}

RadioField.propTypes = {
  ...CheckField.propTypes,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
}
