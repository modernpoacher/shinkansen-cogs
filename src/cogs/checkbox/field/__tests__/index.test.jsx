import React, { Component as mockComponent } from 'react'
import renderer from 'react-test-renderer'
import classnames from 'classnames'

import { CheckField } from '#cogs/components/field'
import Field from '#cogs/cogs/checkbox/field'

jest.mock('classnames', () => jest.fn(() => 'MOCK CLASSNAME'))

jest.mock('#cogs/components/field', () => {
  class MockField extends mockComponent {
    getClassName () {
      return 'MOCK CLASSNAME'
    }

    handleChange () { }

    handleClick () { }
  }

  return {
    __esModule: true,
    CheckField: class CheckField extends MockField { },
    default: MockField
  }
})

describe('#cogs/cogs/checkbox/field', () => {
  describe('<Field />', () => {
    describe('With required props', () => {
      const component = (
        <Field
          name='MOCK NAME'
          value='MOCK VALUE'
        />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('Prototype', () => {
        describe('`getClassName`', () => {
          it('is defined', () => {
            return expect(Field.prototype.getClassName)
              .toBeDefined()
          })
        })
      })

      describe('Instance', () => {
        let instance

        beforeEach(() => {
          instance = (
            renderer.create(component)
              .getInstance()
          )
        })

        describe('`handleClick`', () => {
          it('is defined', () => {
            return expect(instance.handleClick)
              .toBeDefined()
          })
        })

        describe('`handleChange`', () => {
          it('is defined', () => {
            return expect(instance.handleChange)
              .toBeDefined()
          })
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Field
            id='MOCK ID'
            name='MOCK NAME'
            title='MOCK TITLE'
            value='MOCK VALUE'
            required
            disabled
            readOnly
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      let returnValue

      beforeEach(() => {
        jest.spyOn(CheckField.prototype, 'getClassName').mockReturnValue('MOCK GETCLASSNAME')

        const component = (
          <Field
            name='MOCK NAME'
            value='MOCK VALUE'
          />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        returnValue = instance.getClassName()
      })

      it('invokes `classnames`', () => {
        return expect(classnames)
          .toBeCalledWith('MOCK GETCLASSNAME', 'checkbox')
      })

      it('returns the classname', () => {
        return expect(returnValue)
          .toBe('MOCK CLASSNAME')
      })
    })

    describe('`handleClick()`', () => {
      it('invokes the `onClick` prop', () => {
        const MOCK_ON_CLICK = jest.spyOn(CheckField.prototype, 'handleClick')

        const component = (
          <Field
            name='MOCK NAME'
            value='MOCK VALUE'
            onClick={MOCK_ON_CLICK}
          />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        instance.handleClick({ target: { value: 'MOCK VALUE', checked: 'MOCK CHECKED' } })

        return expect(MOCK_ON_CLICK)
          .toBeCalledWith({ target: { value: 'MOCK VALUE', checked: 'MOCK CHECKED' } })
      })
    })

    describe('`handleChange()`', () => {
      it('invokes the `onChange` prop', () => {
        const MOCK_ON_CHANGE = jest.spyOn(CheckField.prototype, 'handleChange')

        const component = (
          <Field
            name='MOCK NAME'
            value='MOCK VALUE'
            onChange={MOCK_ON_CHANGE}
          />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        instance.handleChange({ target: { value: 'MOCK VALUE', checked: 'MOCK CHECKED' } })

        return expect(MOCK_ON_CHANGE)
          .toBeCalledWith({ target: { value: 'MOCK VALUE', checked: 'MOCK CHECKED' } })
      })
    })
  })
})
