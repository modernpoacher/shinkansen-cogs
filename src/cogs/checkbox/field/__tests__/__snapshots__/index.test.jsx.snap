// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#cogs/cogs/checkbox/field <Field /> With additional props renders 1`] = `
<input
  className="MOCK CLASSNAME"
  disabled={true}
  id="MOCK ID"
  name="MOCK NAME"
  onChange={[Function]}
  onClick={[Function]}
  readOnly={true}
  required={true}
  type="checkbox"
  value="MOCK VALUE"
/>
`;

exports[`#cogs/cogs/checkbox/field <Field /> With required props renders 1`] = `
<input
  className="MOCK CLASSNAME"
  disabled={false}
  name="MOCK NAME"
  onChange={[Function]}
  onClick={[Function]}
  readOnly={false}
  required={false}
  type="checkbox"
  value="MOCK VALUE"
/>
`;
